sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -0,0 +1,351 @@
+
+    switch (event) {
+        case NETDEV_EVENT_MLME_INDICATION:
+            /* ignore */
+            break;
+        case NETDEV_EVENT_MCPS_INDICATION:
+            mcps_indication = mac->mcps_buf;
+            if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_LORAWAN, mcps_indication->data.port, mcps_indication->data.pkt)) {
+                gnrc_pktbuf_release(mcps_indication->data.pkt);
+            }
+            break;
+        case NETDEV_EVENT_MLME_CONFIRM:
+            _mlme_confirm((gnrc_netif_t *) mac->netdev.context, mac->mlme_buf);
+            break;
+        case NETDEV_EVENT_MCPS_CONFIRM:
+            mcps_confirm = mac->mcps_buf;
+            if (mcps_confirm->status == 0) {
+                gnrc_pktbuf_release(mac->mcps.outgoing_pkt);
+            }
+            else {
+                gnrc_pktbuf_release_error(mac->mcps.outgoing_pkt, 1);
+            }
+            mac->mcps.outgoing_pkt = NULL;
+            break;
+        case NETDEV_EVENT_MLME_GET_BUFFER:
+            mac->mlme_buf = _mlme_buffer;
+            break;
+        case NETDEV_EVENT_MCPS_GET_BUFFER:
+            mac->mcps_buf = _mcps_buffer;
+            break;
+        default:
+            netdev_event_cb_pass(dev, event);
+    }
+}
+
+static void _driver_cb(netdev_t *dev, netdev_event_t event)
+{
+    gnrc_lorawan_t *mac = (gnrc_lorawan_t *) dev->context;
+    gnrc_netif_t *netif = (gnrc_netif_t *) mac->netdev.context;
+
+    if (event == NETDEV_EVENT_ISR) {
+        msg_t msg = { .type = NETDEV_MSG_TYPE_EVENT,
+                      .content = { .ptr = netif } };
+
+        if (msg_send(&msg, netif->pid) <= 0) {
+            DEBUG("gnrc_netif: possibly lost interrupt.\n");
+        }
+    }
+    else {
+        DEBUG("gnrc_netif: event triggered -> %i\n", event);
+        switch (event) {
+            case NETDEV_EVENT_RX_COMPLETE:
+                gnrc_lorawan_recv(mac);
+                break;
+            case NETDEV_EVENT_TX_COMPLETE:
+                gnrc_lorawan_event_tx_complete(mac);
+                break;
+            case NETDEV_EVENT_RX_TIMEOUT:
+                gnrc_lorawan_event_timeout(mac);
+                break;
+            default:
+                DEBUG("gnrc_netif: warning: unhandled event %u.\n", event);
+        }
+    }
+}
+
+static void _reset(gnrc_netif_t *netif)
+{
+    netif->lorawan.otaa = LORAMAC_DEFAULT_JOIN_PROCEDURE == LORAMAC_JOIN_OTAA ? NETOPT_ENABLE : NETOPT_DISABLE;
+    netif->lorawan.datarate = LORAMAC_DEFAULT_DR;
+    netif->lorawan.demod_margin = 0;
+    netif->lorawan.num_gateways = 0;
+    netif->lorawan.port = LORAMAC_DEFAULT_TX_PORT;
+    netif->lorawan.ack_req = LORAMAC_DEFAULT_TX_MODE == LORAMAC_TX_CNF;
+    netif->lorawan.flags = 0;
+}
+
+static void _memcpy_reversed(uint8_t *dst, uint8_t *src, size_t size)
+{
+    for(size_t i=0;i<size;i++) {
+        dst[size-i-1] = src[i];
+    }
+}
+
+static void _init(gnrc_netif_t *netif)
+{
+    netif->dev->event_callback = _driver_cb;
```

Comment:
It might be recommendable to add a call to `gnrc_netif_init_default()` here, no that #10485 is merged.

Reply:
you are right, I will.

Reply:
addressed in https://github.com/RIOT-OS/RIOT/pull/11022/commits/60df21c68451fa5a8374c5bca5bdf5ebf7d8486f.